streams/src/test/java/org/apache/kafka/streams/kstream/MaterializedTest.java

```diff
@@ -51,4 +54,21 @@ public void shouldThrowNullPointerIfKeyValueBytesStoreSupplierIsNull() {
     public void shouldThrowNullPointerIfSessionBytesStoreSupplierIsNull() {
         Materialized.as((SessionBytesStoreSupplier) null);
     }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldThrowIllegalArgumentExceptionIfRetentionIsNegative() {
+        Materialized.as("valid-name").withRetention(Duration.of(-1, ChronoUnit.DAYS));
```

Comment:
While existing test are written this way, we try to move off this pattern and not use the `expected` annotation. Instead, we should use `assertThrows` and also verify the exception error message.

Reply:
Used assertThrows instead of expected annotation. Validated exception error message as well.